src/frontend/screens/Achievements/GameAchievementDetails/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { GameAchievements } from '@hyperplay/ui'
+import React, { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { achievementsSortOptions } from '..'
+import walletState from 'frontend/state/WalletState'
+import { useTranslation } from 'react-i18next'
+import { observer } from 'mobx-react-lite'
+import AchievementStoreState from 'frontend/state/AchievementStoreState'
+import MintAchievementsState from 'frontend/state/MintAchievementsState'
+
+const pageSize = 6
+
+function isTimestampInPast(unixTimestamp: number) {
+  const currentTime = new Date().getTime()
+  const timestampInMilliseconds = unixTimestamp * 1000 // Convert to milliseconds
+
+  return timestampInMilliseconds < currentTime
+}
+
+export default observer(function GameAchievementDetails(): JSX.Element {
+  const { t } = useTranslation()
+  const { id } = useParams()
+
+  const [selectedSort, setSelectedSort] = useState(achievementsSortOptions[0])
+
+  const achievementsToBeMinted = MintAchievementsState.achievementsToBeMinted
+  const isLoading = MintAchievementsState.isLoading
+  const handleMint = MintAchievementsState.handleMint
+  const handleUpdate = MintAchievementsState.handleUpdate
+  const achievementsToBeUpdated = MintAchievementsState.achievementsToBeUpdated
+
+  const numFreeMints = AchievementStoreState.numFreeMints
+  const individualAchievements = AchievementStoreState.individualAchievements
+  const summaryAchievement = AchievementStoreState.summaryAchievements.data[0]
```

Comment:
**Optional** but perhaps simplifies and less lines 

```typescript
  const {
    achievementsToBeMinted,
    achievementsToBeUpdated,
    isLoading,
    handleMint,
    handleUpdate,
  } = MintAchievementsState;

  const {
    numFreeMints,
    individualAchievements,
    summaryAchievements,
  } = AchievementStoreState;
```

Reply:
sure thing. updated